lib/unique_elements.rb

```diff
@@ -1,6 +1,26 @@
 # This method takes an array of elements (these might be strings, integers, floats, or a
 # combination of the above), and returns an array of the elements that appear once and only
 # once.
-def find_unique_elements(arr)
+
+def count_elements(arr)
+  h={}
+  arr.each do |n|
+    if h.has_key?(n) == false
+      h[n]=1
+    elsif h.has_key?(n) == true
+      h[n]= h[n] +1
+    end
+  end
+  return h
+end  
   
+def find_unique_elements(arr)
```

Comment:
This method is missing an `end` somewhere. It can be easier to see these for yourself if you indent your code meticulously (which I will admit can take some practice to get used to).

It's easiest if you think about indentation as something you do whenever a particular line of code isn't considered "complete" by the end of the line. For example, `x = "Sumeet"` is a complete line of code, whereas `def throw_party(invitees)` is not (because `def` means that we're beginning to write a method, but the method body isn't on that line).

Or another example is `guest_list.each do |invitee|`. We're about to begin a block of code to run on each `invitee`, so clearly this line is just the beginning of some operation.

If you indent your code accordingly, you can then usually spot a missing `end` just by eyeballing it–you'll notice the symmetry is off.

Like if you had this code:

```ruby
def throw_party(invitees)
    guest_list = []
    invitees.each do |guest|
        guest_list << invitee
        guest.family_members.each do |fam|
            guest_list << fam
        end
end
```

Without even understanding the code itself (Because the code itself is kind of nonsensical), you should be able to eyeball the missing `end` by seeing where there is too much horizontal space between one block of code and the line that follows it.

Let me know if I should elaborate.